code/game/objects/items/devices/multitool.dm

```diff
@@ -35,36 +40,54 @@
 	return ..()
 
 /obj/item/device/multitool/ai_detect/process()
-	if(track_delay > world.time)
+	if(track_cooldown > world.time)
 		return
 
-	var/found_eye = 0
-	var/turf/our_turf = get_turf(src)
+	detect_state = 0
+	our_turf = get_turf(src)
+	multitool_detect()
 
+	if(detect_state == 1)

```

Comment:
if you wanted to go the EXTRA mile, I wouldn't use numbers for the defines, I'd do:
```
#define DETECTOR_VERY_NEAR "_red"
#define DETECTOR_NEAR "_yellow"
#define DETECTOR_NONE ""
```
then this if/elseif/else block can just be ```icon_state = "[initial(icon_state)][detect_state]"```